CMakeLists.txt

```diff
@@ -16,20 +16,29 @@ endif()
 
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 
+include(GenerateExportHeader)
 add_library(rsl
     src/parameter_validators.cpp
     src/random.cpp
 )
 add_library(rsl::rsl ALIAS rsl)
 target_compile_features(rsl PUBLIC cxx_std_17)
-target_include_directories(rsl PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
+target_include_directories(rsl PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
+)
 target_link_libraries(rsl PUBLIC
     Eigen3::Eigen
     fmt::fmt
     rclcpp::rclcpp
     tcb_span::tcb_span
     tl_expected::tl_expected
 )
+set_target_properties(rsl PROPERTIES CXX_VISIBILITY_PRESET hidden VISIBILITY_INLINES_HIDDEN YES)
+generate_export_header(rsl EXPORT_FILE_NAME include/rsl/export.hpp)
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(rsl PUBLIC RSL_STATIC_DEFINE)
+endif()
```

Comment:
Are you sure that this is required? If `rsl` is a static library, then the `generate_export_header` already make sure that the `RSL_EXPORT` and `RSL_IMPORT` macros are empty, i.e. this is what you obtaine on Windows if `BUILD_SHARED_LIBS` is `ON`:

~~~
#ifdef RSL_STATIC_DEFINE
#  define RSL_EXPORT
#  define RSL_NO_EXPORT
#else
#  ifndef RSL_EXPORT
#    ifdef rsl_EXPORTS
        /* We are building this library */
#      define RSL_EXPORT __declspec(dllexport)
#    else
        /* We are using this library */
#      define RSL_EXPORT __declspec(dllimport)
#    endif
#  endif

#  ifndef RSL_NO_EXPORT
#    define RSL_NO_EXPORT 
#  endif
#endif
~~~

and this if it is OFF:
~~~
#ifdef RSL_STATIC_DEFINE
#  define RSL_EXPORT
#  define RSL_NO_EXPORT
#else
#  ifndef RSL_EXPORT
#    ifdef rsl_EXPORTS
        /* We are building this library */
#      define RSL_EXPORT 
#    else
        /* We are using this library */
#      define RSL_EXPORT 
#    endif
#  endif

#  ifndef RSL_NO_EXPORT
#    define RSL_NO_EXPORT 
#  endif
#endif
~~~

Reply:
Thanks for letting me know! I now see in the docs why that is the case.